bin/templates/cordova/lib/builders/ProjectBuilder.js

```diff
@@ -365,6 +382,31 @@ function findOutputApksHelper (dir, build_type, arch) {
     return ret;
 }
 
+function findOutputBundlesHelper (dir, build_type) {
+    var shellSilent = shell.config.silent;
+    shell.config.silent = true;
+
+    // list directory recursively
+    var ret = shell.ls('-R', dir).map(function (file) {
```

Comment:
I just wanted to make a note that we been trying to drop `shelljs` as a dependency across repos when possible.

https://issues.apache.org/jira/browse/CB-14140

Reply:
Good to know.. I was mimicking what was already in the codebase for this. Is there a replacement for `shelljs`?

Reply:
> https://issues.apache.org/jira/browse/CB-14140

You may want to check out apache/cordova-common#21 & apache/cordova-lib#613 where we did the replacement on a couple major modules.

FYI We did encounter apache/cordova#121 on Windows, which was eventually solved by a workaround on fs-extra@8.

And for spawning external processes, we migrated from Node.js “child_process” to `cross-spawn` in apache/cordova-common#50, started discussing further work in apache/cordova#16.

I am thinking we should stick with `shelljs` for a minor release, to resolve the issue, and consider migrating to `fs-extra` in the next major release, to avoid an accidentally breaking change.

> Dropping `shelljs` I think would make more sense to be done as a separate PR?

That would be my vote.

Reply:
> Dropping shelljs I think would make more sense to be done as a separate PR?

and

> I am thinking we should stick with shelljs for a minor release, to resolve the issue, and consider migrating to fs-extra in the next major release, to avoid an accidentally breaking change.

Dropping the usage of `shelljs` that already exists would be in a separate PR from this ticket and would be major change as @brodybits pointed out, except what I am pointing at is brand new code that is introduced in this PR that could be avoided from being added.

It would not be a breaking change.

Anyways, you can keep it in, if you prefer, and we remove it later in the next major. The next major will deprecates node 6 and 8 which then we could a cleaner recursive directory scan.

Example: https://stackoverflow.com/a/45130990/1644129

Reply:
I think it would be best to wait so that the `shelljs` refactor can be done all at once. Otherwise I'd have to include new dependencies and some of the code base will be using the new replacement while other parts will be still using `shelljs`.